### PR TITLE
Option to keep deck playing on track load

### PR DESCRIPTION
**Pull request by mbacarella**
This is a proposed solution as discussed in issue #10548.

This is my first proposed contribution to this project and first to an open source C++ project. I apologize if I missed something important or did something dumb. Thanks for your consideration :)

### CODE REVIEW

**Comment by ronso0** [comment]
Thank you for working on this! And congratulations for your first contribution : )

Since this is your first contribution we need you to sign the [contributor agreement](https://docs.google.com/forms/d/e/1FAIpQLScC9QG327sjLL0eWftmfGUasxFFLxg6LCXJ2xHDYRzFIRqyiw/viewform?formkey=dEpYN2NkVEFnWWQzbkFfM0ZYYUZ5X2c6MQ) and thereby allows Mixxx to include your changes.
Please comment here when you filled the form and let us know which name should appear in the contributor list.
> I apologize if I missed something important

Please install pre-commit as described in [CONTRIBUTING.md#git-workflow](https://github.com/mixxxdj/mixxx/blob/main/CONTRIBUTING.md#git-workflow) (I think [wiki/Using-Git#set-up-automatic-code-checking](https://github.com/mixxxdj/mixxx/wiki/Using-Git#set-up-automatic-code-checking) is more comprehensive)

**Comment by ronso0** [comment]
> Please install pre-commit as described in [CONTRIBUTING.md#git-workflow](https://github.com/mixxxdj/mixxx/blob/main/CONTRIBUTING.md#git-workflow) (I think [wiki/Using-Git#set-up-automatic-code-checking](https://github.com/mixxxdj/mixxx/wiki/Using-Git#set-up-automatic-code-checking) is more comprehensive)

As long as no one started to review this PR you may as well incorporate the pre-commit fixes and sqash everything into one commit and rebase onto mixxx/main.

**Comment by mbacarella** [comment]
> Please comment here when you filled the form and let us know which name should appear in the contributor list.

I filled out the form. `Michael Bacarella <m@bacarella.com>` is fine.

**Comment by mbacarella** [comment]
>As long as no one started to review this PR you may as well incorporate the pre-commit fixes and sqash everything into one commit and rebase onto mixxx/main.

I installed the pre-commit hooks, squashed, rebased and pushed. It complained about `qmlformat` but I couldn't figure out how to install it, so I skipped. I assume the server side may run it and let me know if there are any problems.

**Comment by mbacarella** [comment]
I apologize for missing the instructions on installing Git pre-commit hooks the first go-round.

**Comment by mbacarella** [comment]
After the rebase, I noticed and fixed two more issues as separate commits and pushed. I thought the clang-format pre-commit hook would tell me to fix things but it passed for me locally without complaint. Not sure what to expect.

File: src/preferences/dialog/dlgprefdeckdlg.ui
Comment (ronso0) [inline src/preferences/dialog/dlgprefdeckdlg.ui:32]: I'm not sure why you changed this (changing translation texts requires new translation for every language).

Also, I thought the new option would a) load a track into a playing deck and b) play that right away.
Even though b) requires a), a) and b) are unrelated (I might want to load tracks into playing decks, but I don not want auto-play)

**Comment by ronso0** [comment]
If did `git rebase -i [mixxx repo]/main` [Catch up to main (resolve merge conflict)](https://github.com/mixxxdj/mixxx/pull/10944/commits/1c7940309ec3b5bceebcb4265a5356c29f99c9f6)
[1c79403](https://github.com/mixxxdj/mixxx/pull/10944/commits/1c7940309ec3b5bceebcb4265a5356c29f99c9f6) would not be necessary (you didn't touch legacyskinparser).

File: src/preferences/dialog/dlgprefdeckdlg.ui
Comment (mbacarella) [reply src/preferences/dialog/dlgprefdeckdlg.ui:32]: It's actually deleting the checkbox entirely and replacing it with a combobox that offers more functionality:

1. load track into playing deck and play that right away
2. load track into playing deck but stop
3. don't allow loading into playing deck

See the discussion at the end of this issue? https://github.com/mixxxdj/mixxx/issues/10548#issuecomment-1268685304

**Comment by mbacarella** [comment]
> If did git rebase -i [mixxx repo]/main Catch up to main (resolve merge conflict) 1c79403 would not be necessary (you didn't touch legacyskinparser).

I can rebase this now, but I notice you have started a review.

File: src/preferences/dialog/dlgprefdeckdlg.ui
Comment (ronso0) [reply src/preferences/dialog/dlgprefdeckdlg.ui:32]: I should have taken a closer look, not just skimming through...

**Comment by ronso0** [comment]
Even though I explicitely just added a comment it is marked as "reviewed", this is irritating.
Feel free to rebase!

File: src/mixer/playermanager.cpp
Comment (daschuer) [inline src/mixer/playermanager.cpp:694]:
```suggestion
    LoadWhenDeckPlaying loadWhenDeckPlaying =
            m_pConfig->getValue(kConfigKeyLoadWhenDeckPlaying, kDefaultLoadWhenDeckPlaying);
```

File: src/mixer/playermanager.h
Comment (daschuer) [inline src/mixer/playermanager.h:191]: The default play in the two functions above should be probably removed to make it explicit.

File: src/preferences/dialog/dlgprefdeck.cpp
Comment (daschuer) [inline src/preferences/dialog/dlgprefdeck.cpp:176]: Do we need to use the default as well, if the value does not exists?

File: src/preferences/dialog/dlgprefdeck.cpp
Comment (daschuer) [inline src/preferences/dialog/dlgprefdeck.cpp:190]: slotSetLoadWhenDeckPlaying()  fixes the relation of combobox index to value. Lets rename it to 
slotLoadWhenDeckPlayingIndexChanged and do the mapping in source code.

File: src/preferences/dialog/dlgprefdeck.cpp
Comment (mbacarella) [reply src/preferences/dialog/dlgprefdeck.cpp:176]: Sorry, I don't follow. If it doesn't exist, won't it then get the default from the else-else?

File: src/preferences/dialog/dlgprefdeck.cpp
Comment (mbacarella) [reply src/preferences/dialog/dlgprefdeck.cpp:190]: Sorry, I don't follow this either.

I just did what the other combo boxes do (e.g. `slotSetTrackLoadMode`). Is there an example of your suggestion? :)

File: src/preferences/dialog/dlgprefdeck.cpp
Comment (daschuer) [reply src/preferences/dialog/dlgprefdeck.cpp:176]: Yes, right. 
It is just the issue that "m_pConfig->exists" is a redundant lookup of the value. It is better to use getValue() with a fallback value like purposed above

File: src/preferences/dialog/dlgprefdeck.cpp
Comment (daschuer) [reply src/preferences/dialog/dlgprefdeck.cpp:190]: It looks like it is sufficient to just rename the function, because it is not a classic setter. It receives the index, fetches the value and than sets the the value. The current name does not expose that an index is expected.

File: src/preferences/dialog/dlgprefdeck.cpp
Comment (mbacarella) [reply src/preferences/dialog/dlgprefdeck.cpp:176]: Well yes but I'm trying to be mindful of not losing behavior previously configured by the user in a previous version. If the user disabled (or enabled) playing track protection I want to import that.

Comment (mbacarella) [reply src/preferences/dialog/dlgprefdeck.cpp:176]: I realize this is confusing because, in a different part of the code, I didn't use `getValue` with default when I should have, but I think in this case it makes sense to probe for existence of the new config variable first and fall back to loading the older config variable.

File: src/preferences/dialog/dlgprefdeck.cpp
Comment (ronso0) [inline src/preferences/dialog/dlgprefdeck.cpp:173]: IMO it makes more sense to have the current default at the top (= the least invasive)
* Reject
* Allow, stop deck
* Allow, play from load point

**Comment by mbacarella** [comment]
Did a bunch of the code review driven changes. *Didn't* rebase since a review actually started after @ronso0 said to rebase away :)

Anything else?

**Comment by ronso0** [comment]
Just a hint: usually code conversations (conversations pinned to a specific spot in the code) are marked as resolved by the reviewer. That makes it easier to check if/how the discussed changes were applied.

**Comment by mbacarella** [comment]
> Just a hint: usually code conversations (conversations pinned to a specific spot in the code) are marked as resolved by the reviewer. That makes it easier to check if/how the discussed changes were applied.

My bad! Will bear that in mind for next time :)

**Comment by mbacarella** [comment]
@ronso0 just to be clear I think this is ready to merge if there are no other blockers

(not sure what to do about the code style check; should I run clang-format on my code and push?)

**Comment by ronso0** [comment]
I only che.cked the UI until now, @daschuer has the final word re code here.

Update pre-commit maybe? Idk why it didn't complain about the indentations locally.
You may also download the generated patch and apply it manually. Click on Details next to the pre-commit check, on the top left click Summary, scroll down to Artifacts and grab [pre-commit.patch](https://github.com/mixxxdj/mixxx/suites/8719138046/artifacts/394105136)

**Review by daschuer — COMMENTED** [review]
I have just tested this and it works good. Thank you. 
Just one comment for improvement and the pre-commit patch and we can merge this nice addition.

File: src/util/dnd.cpp
Comment (mbacarella) [reply src/util/dnd.cpp]: Done

**Review by daschuer — APPROVED** [review]
LGTM, Thank you.

**Pull request merged**
